server/storage.go

```diff
@@ -1,27 +1,65 @@
 package server
 
+import (
+	"fmt"
+)
+
 type User struct {
 	ID      string `json:"id"`
 	Name    string `json:"name"`
 	Surname string `json:"surname"`
 	Age     int    `json:"age"`
 }
 
-var storage []User
-
 type Storage interface {
 	GetUser(id string) (*User, error)
 	CreateUser(*User) error // Make sure ID is added to the struct OR return ID from this method
 	DeleteUser(id string) error
+	GetAllUsers() (*[]User, error)
+}
+
+type UserStorage struct {
+	storage []User
+}
+
+func (us *UserStorage) GetAllUsers() (*[]User, error) {
+	return &us.storage, nil
+}
+
+func (us *UserStorage) GetUser(id string) (*User, error) {
+	for _, user := range us.storage {
+		if user.ID == id {
+			return &user, nil
+		}
+	}
+	return nil, fmt.Errorf("User with id %v was not found", id)
+}
+
+func (us *UserStorage) CreateUser(*User) error {
+	user := User{}
```

Comment:
This method is supposed to create a user that is received as an argument. We receive a user in JSON, parse this JSON into the struct in handler and then provide it to storage to store